fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -460,15 +570,20 @@ private void setDateValues(Parameter p, java.time.Instant start, java.time.Insta
         Timestamp implicitEndInclusive = convertToInclusiveEnd(implicitEndExclusive);
         p.setValueDateEnd(implicitEndInclusive);
     }
-    
+
     /**
      * Convert a period's end timestamp from an exclusive end timestamp to an inclusive one
-     * 
+     *
      * @param exlusiveEndTime
      * @return inclusiveEndTime
      */
     private Timestamp convertToInclusiveEnd(Timestamp exlusiveEndTime) {
         // Our current schema uses the db2/derby default of 6 decimal places (1000 nanoseconds) for fractional seconds.
         return Timestamp.from(exlusiveEndTime.toInstant().minusNanos(1000));
     }
+
+    private IllegalArgumentException invalidComboException(SearchParamType paramType, Element value) {
+        return new IllegalArgumentException("Data type '" + value.getClass().getSimpleName() + "' is not supported "
+                + "for SearchParameter of type '" + paramType.getValue() + "'");
```

Comment:
I think the Search Parameter should be added here... the combination will help provide a trajectory to resolve.

Reply:
Also, not a super fan of IllegalArgumentException.   why the choice here?

Reply:
> Also, not a super fan of IllegalArgumentException. why the choice here?
Unless we want to change the visitor interface to throw a checked exception, it needs to be a RunTime exception and IllegalArgumentException is the best fit.

Reply:
```suggestion
                + "for SearchParameter '" + searchParamCode + "' of type '" + paramType.getValue() + "'");
```